RFCs/0014-ToJsonString.md

```diff
@@ -0,0 +1,113 @@
+                        "width": 3,
+                        "height": 3,
+                        "properties": { "markdown": "Hello world" }
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+## YAML
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            Fn::ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+### Short Form Syntax
+```yaml
+MyDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+        DashboardBody:
+            !ToJsonString:
+                start: "-PT6H"
+                periodOverride: inherit
+                widgets:
+                    - type: text
+                    x: 0
+                    y: 7
+                    width: 3
+                    height: 3
+                    properties:
+                        markdown: Hello world
+```
+
+# Motivation
+
+A CloudFormation user may want to use JSON strings as input to a resource property. For example, `AWS::CloudWatch::Dashboard` requires a JSON string for the `DashboardBody` attribute. A user may even want to use a JSON string as input to attributes where only a general string is required, such as in `SecretString` of `AWS::SecretsManager::Secret`.
+
+Today, this can only be accomplished through the use of external tools to convert object bodies to JSON strings, or use workarounds such as YAML multiline syntax. Having a feature to automate this transformation can provide various benefits, such as improving development workflow, improving code readability, or being able to utilize JSON/YAML syntax highlighting in text editors.
+
+Open Github request regarding this: https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/78
+
+# Details
+
+`Fn::ToJsonString` is an intrinsic function that takes in a template block as input and converts it into an escaped JSON string.
+
+* It will be restricted to only be used as the value of string-type resource properties.
+* Intrinsic functions (e.g. `Fn::If`, `Ref`) or pseudo parameters (e.g. `AWS::NoValue`) can be used within the input template block. The input template block will be processed by the intrinsic functions and pseudo parameters before it is converted to a string (i.e. the resolved value of the intrinsic function is converted to the output string, and not the intrinsic function itself).
+* Conversion will always retain the same order of key-value pairs such that the converted strings of the same input template block are guaranteed to not change.
+* The following pseudo parameters will be supported:
+    * `AWS::AccountId`, `AWS::Region`, `AWS::Partition`, `AWS::NoValue`
+
+# Limitation
```

Comment:
What is the cause of these limitations?

Reply:
These are short-term limitations due to implementation constraints.